source/kernel/src/lib.rs

```diff
@@ -240,11 +253,20 @@ impl Kernel {
             }
         }
 
-        inner.scheduler.tick();
+        inner.scheduler.tick()
 
         // TODO: Send time to userspace?
     }
 
+    /// Initialize the kernel's `maitake` timer as the global default timer.
+    ///
+    /// This allows the use of `sleep` and `timeout` free functions.
+    /// TODO(eliza): can the kernel just "do this" once it becomes active? Or,
+    /// have a "kernel.init()" or something that does this and other global inits?
```

Comment:
ATM, the only "init" is what we do in `new`. There are a couple steps that are "place the global thing", then use it.

We could document that `Kernel::new()` should only be called once.

FWIW: I haven't really been using any globals, and we might want to consider just having a kernel global that gives access to everything.

Reply:
I would love to have a kernel global eventually --- I'm thinking this would be rolled into whatever sets up the global kernel context, when that exists. I don't think `Kernel::new` can do this, because the `set_global_timer` interface is only callable with a `&'static Timer`...